src/crash_screen.c

```diff
@@ -83,120 +83,38 @@ void crash_screen_draw_glyph(u16 *framebuffer, s32 x, s32 y, s32 glyph) {
-                    if (temp_t0 != 0) {
-                        phi_v0 = temp_ra + (phi_s2 * 2);
-                        do { // (temp_t0 + j;) {
-                            j++;
-                            //temp_v1 = phi_v1 + 1;
-                            //phi_v1 = temp_v1;
-                            //phi_v1_2 = temp_v1;
-                            if (i - j == 6) {
-                                framebuffer[i * 640 + j * 2] = RED_COLOUR;
-                            } else {
-                                framebuffer[i * 640 + j * 2] = WHITE_COLOUR;
-                            }
-                            j += 2;
-                        } while ((temp_t0 + j) != j);
-                        phi_a1_4 = temp_v1 * 2;
-                        if (temp_v1 != phi_s0) {
-                            break; // goto 10
-                        }
-                    } else {
-//block_10:
-                        //temp_a1 = phi_v1_2 * 2;
-                        phi_v0_2 = framebuffer[i * 640 + j * 2];
-                        phi_a1 = temp_a1;
-                        for (k = j + 8; k < j * 2; k += 8) {
-                        //do {
-                            //temp_a1_2 = phi_a1 + 8;
-                            //phi_a1 = temp_a1_2;
-                            //phi_a1_4 = temp_a1_2;
-                            if (temp_a0 == 6) {
-                                framebuffer[i * 640 + j * 2] = RED_COLOUR;
-                            } else {
-                                framebuffer[i * 640 + j * 2] = WHITE_COLOUR;
-                            }
-                            //if (temp_a0 == 6) {
-                            //    phi_v0_2->unk2 = RED_COLOUR;
-                            //} else {
-                            //    phi_v0_2->unk2 = WHITE_COLOUR;
-                            //}
-                            //if (temp_a0 == 6) {
-                            //    phi_v0_2->unk4 = RED_COLOUR;
-                            //} else {
-                            //    phi_v0_2->unk4 = WHITE_COLOUR;
-                            //}
-                            //if (temp_a0 == 6) {
-                            //    phi_v0_2->unk6 = RED_COLOUR;
-                            //} else {
-                            //    phi_v0_2->unk6 = WHITE_COLOUR;
-                            //}
-                            //phi_v0_2 += 8;
-                        } //while (temp_a1_2 != (phi_s0 * 2));
-                    }
+    s32 var_s0;
+    s32 var_s2;
+    s32 row;
+    s32 column;
+
+    for (var_s0 = 0x2E, var_s2 = 0x28; var_s0 > 0x2C; var_s0--, var_s2++) {
+        for (row = var_s2; row < var_s0; row++) {
+            for (column = var_s2; column < var_s0; column++) {
+                if ((var_s0 - var_s2) == 6) {
+                    framebuffer[(row * 320) + column] = RED_COLOUR;
+                } else {
+                    framebuffer[(row * 320) + column] = WHITE_COLOUR;
```

Comment:
Can you update the func to align with our latest diff:
(change func prototype and `framebuffer[row * 320 + column] =`
https://decomp.me/scratch/pUilB

and then I'll merge.

Reply:
I think its currently up-to-date with that? Prototype seems right and the array access is as you commented.

Reply:
Oh, my bad! I didn't look that closely apparently.